astropy/time/core.py

```diff
@@ -437,6 +438,9 @@ def precision(self, val):
             raise ValueError('precision attribute must be an int between '
                              '0 and 9')
         self._precision = val
+        if hasattr(self, '_time'):
```

Comment:
p.s. Independently, yes, the cache should be deleted -- unless we can somehow use in-place changes such that all `_precision` attributes always refers to the same object (but probably too much risk of foot-shooting).